Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift

```diff
@@ -165,15 +189,269 @@ struct IDEDiagnosticConsoleFormatter: DiagnosticConsoleFormatter {
+    
+    init(
+        baseUrl: URL?,
+        highlight: Bool,
+        options: DiagnosticFormattingOptions
+    ) {
+        self.baseUrl = baseUrl
+        self.highlight = highlight
+        self.options = options
     }
     
+    func formattedDescription<Problems>(for problems: Problems) -> String where Problems: Sequence, Problems.Element == Problem {
+        let sortedProblems = problems.sorted { lhs, rhs in
+            guard let lhsSource = lhs.diagnostic.source,
+                  let rhsSource = rhs.diagnostic.source
+            else { return lhs.diagnostic.source  == nil }
+            
+            guard let lhsRange = lhs.diagnostic.range,
+                  let rhsRange = rhs.diagnostic.range
+            else { return lhsSource.path < rhsSource.path }
+            
+            if lhsSource.path == rhsSource.path {
+                return lhsRange.lowerBound < rhsRange.lowerBound
+            } else {
+                return lhsSource.path < rhsSource.path
+            }
+        }
+
+        return sortedProblems.map { formattedDescription(for: $0) }.joined(separator: "\n\n")
+    }
+
+    func formattedDescription(for problem: Problem) -> String {
+        formattedDiagnosticsSummary(for: problem.diagnostic) +
+        formattedDiagnosticDetails(for: problem.diagnostic) +
+        formattedDiagnosticSource(for: problem.diagnostic, with: problem.possibleSolutions)
+    }
+
     func formattedDescription(for diagnostic: Diagnostic) -> String {
-        return IDEDiagnosticConsoleFormatter(options: options).formattedDescription(for: diagnostic)
+        formattedDiagnosticsSummary(for: diagnostic) +
+        formattedDiagnosticDetails(for: diagnostic) +
+        formattedDiagnosticSource(for: diagnostic, with: [])
+    }
+}
+
+extension DefaultDiagnosticConsoleFormatter {
+    private func formattedDiagnosticsSummary(for diagnostic: Diagnostic) -> String {
+        let summary =  diagnostic.severity.description + ": " + diagnostic.summary
+        if highlight {
+            let ansiAnnotation = diagnostic.severity.ansiAnnotation
+            return ansiAnnotation.applied(to: summary)
+        } else {
+            return summary
+        }
+    }
+    
+    private func formattedDiagnosticDetails(for diagnostic: Diagnostic) -> String {
+        var result = ""
+        if let explanation = diagnostic.explanation {
+            result.append("\n\(explanation)")
+        }
+        
+        if !diagnostic.notes.isEmpty {
+            let formattedNotes = diagnostic.notes
+                .map { note in
+                    let location = "\(formattedSourcePath(note.source)):\(note.range.lowerBound.line):\(note.range.lowerBound.column)"
+                    return "\(location): \(note.message)"
+                }
+                .joined(separator: "\n")
+            result.append("\n\(formattedNotes)")
+        }
+        
+        return result
+    }
+    
+    private func formattedDiagnosticSource(
```

Comment:
minor: This implementation could benefit from a few inline examples of how each part of the source description looks. That would make it easier to understand where in the description the "-->" or the line number prefixed "|" and "+" appear.